presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -452,7 +453,8 @@ public void finishInsertTable(JdbcIdentity identity, JdbcOutputTableHandle handl
     {
         String temporaryTable = quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTemporaryTableName());
         String targetTable = quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTableName());
-        String insertSql = format("INSERT INTO %s SELECT * FROM %s", targetTable, temporaryTable);
+        String columns = join(",", handle.getColumnNames());
+        String insertSql = format("INSERT INTO %s (%s) SELECT %s FROM %s", targetTable, columns, columns, temporaryTable);
```

Comment:
I think the `SELECT *` is ok since we create the temporary table with only the supported columns.